crates/router/src/core/payments/tokenization.rs

```diff
@@ -242,6 +242,16 @@ pub async fn add_payment_method_token<F: Clone, T: Clone>(
                 pm_token_request_data,
                 pm_token_response_data,
             );
+
+            connector_integration
+                .execute_pretasks(&mut pm_token_router_data, state)
+                .await
+                .to_payment_failed_response()?;
+
+            router_data
+                .request
+                .set_session_token(pm_token_router_data.session_token.clone());
```

Comment:
This is not generic, Only for square we are using session_token in Tokenize flow

Reply:
Yes the variable session_token is an option and the function set_session_token will set it to none in all other cases.